projects/kit/components/avatar/tests/avatar.component.spec.ts

```diff
@@ -50,55 +49,53 @@ describe(`Avatar`, () => {
 
     beforeEach(() => {
         fixture = TestBed.createComponent(TestComponent);
-        pageObject = new TuiPageObject(fixture);
-        testComponent = fixture.componentInstance;
-        fixture.detectChanges();
-        component = testComponent.component;
+        loader = TestbedHarnessEnvironment.loader(fixture);
     });
 
     describe(`computedText`, () => {
-        it(`if there is an avatar, the text value is empty`, () => {
-            expect(component.computedText).toBe(``);
+        it(`if there is an avatar, the text value is empty`, async () => {
+            const avatar = await loader.getHarness(TuiAvatarHarness);
+
+            expect(await avatar.text()).toBe(``);
         });
 
-        it(`if there is no avatar, the text value is taken from the first letters of the words in text`, () => {
-            testComponent.avatarUrl = null;
-            fixture.detectChanges();
+        it(`if there is no avatar, the text value is taken from the first letters of the words in text`, async () => {
+            const avatar = await loader.getHarness(
+                TuiAvatarHarness.with({selector: `#null-url`}),
+            );
 
-            expect(component.computedText).toBe(`JC`);
+            expect(await avatar.text()).toBe(`JC`);
         });
 
-        it(`if the avatar is absent, and there is one word in text, its first letter is taken`, () => {
-            testComponent.avatarUrl = null;
-            testComponent.text = `James`;
-            fixture.detectChanges();
+        it(`if the avatar is absent, and there is one word in text, its first letter is taken`, async () => {
+            const avatar = await loader.getHarness(
+                TuiAvatarHarness.with({selector: `#null-url-with-text`}),
+            );
 
-            expect(component.computedText).toBe(`J`);
+            expect(await avatar.text()).toBe(`J`);
         });
 
-        it(`for xs sizes only one letter is taken`, () => {
-            testComponent.avatarUrl = null;
-            testComponent.size = `xs`;
-            fixture.detectChanges();
+        it(`for xs sizes only one letter is taken`, async () => {
+            const avatar = await loader.getHarness(
+                TuiAvatarHarness.with({selector: `#null-url-xs`}),
+            );
 
-            expect(component.computedText).toBe(`J`);
+            expect(await avatar.text()).toBe(`J`);
         });
     });
 
     // TODO: Jest doesn't support intersection observe
     xdescribe(`Avatar color`, () => {
         it(`if there is an avatarUrl the color is rgba(0, 0, 0, 0)`, () => {
-            expect(getComputedStyle(getAvatar()).backgroundColor).toBe(
-                `rgba(0, 0, 0, 0)`,
-            );
+            /* expect(getComputedStyle(getAvatar()).backgroundColor).toBe( */
+            /*     `rgba(0, 0, 0, 0)`, */
+            /* ); */
         });
 
         it(`when autoColor is on, the color will be - rgb(160, 170, 228)`, () => {
```

Comment:
`xit` instead of `/* expect(getComputedStyle(getAvatar()).backgroundColor).toBe( */`

Reply:
@zarghamkhandev please see my comment

Reply:
sorry I forgot to push, I have reverted this change.